### PR TITLE
Proposal to handle MSIE11 compatibility

### PR DESCRIPTION
**Pull request by gijskant**
- change target to `es5`
- add polyfills for older browsers
- add message to MSIE users

Probably better approach:
- create two build targets:
  - `es5` for older browsers
  - `es2016` for modern browsers
- include one of them based on which browser. However, conditional includes seem to have disappeared since IE10...

### CODE REVIEW

File: src/app/modules/gb-analysis-module/gb-analysis.component.css [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* top, right, bottom, left */
 .analysis-container {
-  margin-top: var(--gb-navbar-height-extra);
+  margin-top: 52px; /* var(--gb-navbar-height-extra); */
```
Comment (beaugogh) [inline src/app/modules/gb-analysis-module/gb-analysis.component.css]: we need variable for this, for clean code

File: src/app/modules/gb-cohort-selection-module/gb-cohort-selection.component.css [hunk 2]
```diff
@@ -7,7 +7,7 @@
   border-radius: 3px;
   border: 1px solid lightgray;
   padding: 0.5em 1em 0.5em 1em;
-  margin-top: var(--gb-navbar-height-extra);
+  margin-top: 52px; /*var(--gb-navbar-height-extra);*/
```
Comment (beaugogh) [inline src/app/modules/gb-cohort-selection-module/gb-cohort-selection.component.css]: see comment above

File: src/app/modules/gb-export-module/gb-export.component.css [hunk 3]
```diff
@@ -3,7 +3,7 @@
 }
 
 .export-container {
-  margin-top: var(--gb-navbar-height-extra);
+  margin-top: 52px; /* var(--gb-navbar-height-extra); */
```
Comment (beaugogh) [inline src/app/modules/gb-export-module/gb-export.component.css]: as before

File: src/polyfills.ts [hunk 4]
```diff
@@ -19,19 +19,24 @@
  */
 
 /** IE9, IE10 and IE11 requires all of the following polyfills. **/
```
Comment (beaugogh) [inline src/polyfills.ts]: if we do not support IE, why include these polyfills?

File: src/tsconfig.app.json [hunk 5]
```diff
@@ -3,7 +3,7 @@
   "compilerOptions": {
     "outDir": "../out-tsc/app",
     "module": "es2015",
```
Comment (beaugogh) [inline src/tsconfig.app.json]: for naming consistency, this could be adjusted to "es6"

File: tsconfig.json [hunk 6]
```diff
@@ -8,7 +8,7 @@
     "moduleResolution": "node",
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
-    "target": "es2016",
+    "target": "es5",
     "typeRoots": [
       "node_modules/@types"
     ],
```
Comment (beaugogh) [inline tsconfig.json]: for naming consistency,  `"module": "es2015"` could be adjusted to "es6"

File: src/app/modules/gb-analysis-module/gb-analysis.component.css (hunk 1 above)
Comment (gijskant) [reply src/app/modules/gb-analysis-module/gb-analysis.component.css]: I agree. I want to further investigate if we can make something like postcss do this automatically, or use scss instead. The preferred way is indeed to make use of all the great features of the latest languages and transpile for backward compatibility.

File: src/app/modules/gb-navbar-module/gb-navbar.component.css [hunk 7]
```diff
@@ -7,7 +7,7 @@
 }
 
 /* Style the links inside the navigation bar */
-.gb-nav a {
+.gb-nav a, .gb-nav-element {
```
Comment (beaugogh) [inline src/app/modules/gb-navbar-module/gb-navbar.component.css]: not sure what this does, originally this is to style the `a` element in navbar, not the entire navbar element

File: src/app/modules/gb-navbar-module/gb-navbar.component.css [hunk 8]
```diff
@@ -35,7 +35,7 @@
   height: var(--gb-navbar-height);
 }
 
-.gb-nav-right a {
+.gb-nav-right a, .gb-nav-element {
```
Comment (beaugogh) [inline src/app/modules/gb-navbar-module/gb-navbar.component.css]: see comment above

File: src/polyfills.ts (hunk 4 above)
Comment (gijskant) [reply src/polyfills.ts]: We don't fully support it, but if we can make basic features work with these widely used and well supported workarounds, it seems a good solution to have an okay user experience for IE users with minimal effort from our side. It's kind of a tradeoff...

File: src/app/modules/gb-navbar-module/gb-navbar.module.ts [hunk 9]
```diff
@@ -12,15 +12,16 @@ import {GbNavbarComponent} from './gb-navbar.component';
 import {RouterModule} from '@angular/router';
 import {TabMenuModule} from 'primeng/components/tabmenu/tabmenu';
 import {FormsModule} from '@angular/forms';
-import {MessagesModule} from 'primeng/primeng';
+import {MessagesModule, TooltipModule} from 'primeng/primeng';
```
Comment (beaugogh) [inline src/app/modules/gb-navbar-module/gb-navbar.module.ts]: are we using tooltip in navbar?

File: src/polyfills.ts (hunk 4 above)
Comment (beaugogh) [reply src/polyfills.ts]: just that we adding these imports, as well as the lib for detecting IE, further increases the overhead, plus all the material icons and fonts we included in the app, things add up.

Comment (beaugogh) [reply src/polyfills.ts]: ideally we want sth. light when built for production, the lighter the better

Comment (beaugogh) [reply src/polyfills.ts]: and btw, I hate IE, so should everyone else :D

File: src/app/modules/gb-navbar-module/gb-navbar.component.css (hunk 7 above)
Comment (gijskant) [reply src/app/modules/gb-navbar-module/gb-navbar.component.css]: This is to also style other elements in the navbar than anchors. In this case, a warning icon.

File: src/app/modules/gb-navbar-module/gb-navbar.module.ts (hunk 9 above)
Comment (gijskant) [reply src/app/modules/gb-navbar-module/gb-navbar.module.ts]: Yes, to inform users that IE is not supported.

File: src/polyfills.ts (hunk 4 above)
Comment (gijskant) [reply src/polyfills.ts]: I agree with you totally on this one!

**Pull request merged**